substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,655 @@
+		InvalidResult,
+		/// The reward calculation failed due to an internal error
+		FailedRewardCalculation,
+		/// Voting round is over
+		VotingRoundOver,
+		/// This voting round does not exists
+		NoRoundFound,
+		/// Maximum number of projects submission for reward distribution as been reached
+		MaximumProjectsNumber,
+		/// Another project has already been submitted under the same project_id
+		SubmittedProjectId,
+		/// Project batch already submitted
+		BatchAlreadySubmitted,
+		/// Requested vote data do not exist
+		NoVoteData,
+		/// Not enough funds to process the transaction
+		NotEnoughFunds,
+		/// This referendum does not exists
+		ReferendumNotFound,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<SystemBlockNumberFor<T>> for Pallet<T> {
+		fn on_idle(_n: SystemBlockNumberFor<T>, remaining_weight: Weight) -> Weight {
+			Self::on_idle_function(remaining_weight)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// OPF Projects registration
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be AdminOrigin
+		///
+		/// ## Details
+		///
+		/// From this extrinsic only AdminOrigin can register whitelisted projects.
+		///
+		/// ### Parameters
+		/// - `projects_id`: The accounts that might be funded.
+		///
+		/// ### Errors
+		/// - [`Error::<T>::MaximumProjectsNumber`]: Maximum number of project subscriptions reached
+		///  
+		/// ## Events
+		/// Emits [`Event::<T>::Projectslisted`].
+		#[pallet::call_index(1)]
+		#[pallet::weight(<T as Config>::WeightInfo::register_projects_batch(T::MaxProjects::get()))]
+		pub fn register_projects_batch(
+			origin: OriginFor<T>,
+			projects_id: BoundedVec<ProjectId<T>, T::MaxProjects>,
+		) -> DispatchResult {
+			let who = T::SubmitOrigin::ensure_origin(origin.clone())?;
+			// Only 1 batch submission per round
+			let mut round_index = NextVotingRoundNumber::<T>::get();
+
+			// No active round?
+			if round_index == 0 {
+				// Start the first voting round
+				let _round0 = VotingRoundInfo::<T>::new();
+				round_index = NextVotingRoundNumber::<T>::get();
+			}
+
+			let current_round_index = round_index.saturating_sub(1);
+
+			let mut round_infos =
+				VotingRounds::<T>::get(current_round_index).expect("InvalidResult");
+
+			// Check no Project batch has been submitted yet
+			ensure!(!round_infos.batch_submitted, Error::<T>::BatchAlreadySubmitted);
+			round_infos.batch_submitted = true;
+			let round_ending_block = round_infos.round_ending_block;
+
+			// If current voting round is over, start a new one
+			let when = T::BlockNumberProvider::current_block_number();
+			if when >= round_ending_block {
+				// Create a new round.
+				let _new_round = VotingRoundInfo::<T>::new();
+			}
+
+			for project_id in &projects_id {
+				ProjectInfo::<T>::new(project_id.clone());
+
+				// Prepare the proposal call
+				let call = Call::<T>::on_registration { project_id: project_id.clone() };
+				let proposal = Self::create_proposal(who.clone(), call);
```

Comment:
You really only should use `conviction-voting`, then you don't need to create a proposal etc.